### PR TITLE
Vanderbilt Data

### PR DESCRIPTION
**Pull request by ilan-gold**
- Add script for vanderbilt
 - Add `gsutil` command for HTTP2 tiles

### CODE REVIEW

File: fake-files/output-expected/vanderbilt_MxIF/vanderbilt_MxIF.images/vanderbilt_MxIF.tiff.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (mccalluc) [inline fake-files/output-expected/vanderbilt_MxIF/vanderbilt_MxIF.images/vanderbilt_MxIF.tiff.json]: Is there a reason the input `vanderbilt.ome.tif` has one "f" and outputs have two?

File: process.sh [hunk 2]
```diff
@@ -33,12 +33,14 @@ main() {
...
     then
-        echo 'CI: Skip push to AWS'
+        echo 'CI: Skip push to AWS and GCS'
     else
         # Exclude the *HUGE* PNGs in the base directory:
         # The tiles for S3 are in subdirectories;
         # We keep the PNGs around because it takes a long time to generate them.
-        aws s3 cp --exclude "$OUTPUT/*.ome.tif" --recursive "$OUTPUT" s3://"$S3_TARGET"
+        TILES_BASE='vanderbilt_MxIF.images'
+        aws s3 cp --exclude "$OUTPUT/*.ome.tif*" --recursive "$OUTPUT" s3://"$S3_TARGET"
```
Comment (mccalluc) [inline process.sh:42]: Should the first argument below (`$OUTPUT/$TILES_BASE/*.ome.tif*`) match the exclude here? (`$OUTPUT/*.ome.tif*`)

(And if consistency on the the number of "f"s is possible, that would also be good.)

File: process.sh [hunk 3]
```diff
@@ -33,12 +33,14 @@ main() {
...
+        gsutil cp "$OUTPUT/$TILES_BASE/*.ome.tif*" gs://vitessce-demo-data/$TILES_BASE
```
Comment (mccalluc) [inline process.sh]:
```suggestion
        gsutil cp "$OUTPUT/$TILES_BASE/*.ome.tif*" "gs://vitessce-demo-data/$TILES_BASE"
```

File: process.sh [hunk 4]
```diff
@@ -15,7 +15,7 @@ main() {
     INPUT="$FILES/input"
     OUTPUT="$FILES/output"
 
-    for DATASET in linnarsson dries wang cao spraggins; do
+    for DATASET in linnarsson dries wang cao spraggins vanderbilt_MxIF; do
```
Comment (mccalluc) [inline process.sh]: Slight preference to just call it `vanderbilt`... unless you think we'll have `vanderbilt_Something-else` in the near future?

File: scripts/process_vanderbilt_MxIF.sh [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+      SERVER_URL="https://vitessce-vanderbilt-data.storage.googleapis.com/"
+      CMD='docker run --rm
+          -e "SERVER_URL='$SERVER_URL'"
+          -e "PREFIX=vanderbilt_MxIF"
+          -e "PYRAMID_TYPE=tiff"
+          --mount "type=bind,src='$INPUT'/vanderbilt.ome.tif,destination=/input.ome.tif"
+          --mount "type=bind,src='$OUTPUT',destination=/output_dir"
+          --name tiler gehlenborglab/ome-tiff-tiler:v0.0.4'
+      echo "Running: $CMD"
+      # vitessce relies on this naming strategy, whereas the docker image is more general
```
Comment (mccalluc) [inline scripts/process_vanderbilt_MxIF.sh]: Should this be two lines lower? Not sure what it applies to... Capitalize first letter and add a period.

File: scripts/process_vanderbilt_MxIF.sh [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+      echo "Skipping tiling -- output already exists: $TILES_PATH"
+  else
+      mkdir $TILES_PATH
+      if [ -e "$INPUT/vanderbilt.ome.tif" ]
+      then
+        echo "Not copying $INPUT/vanderbilt.ome.tif from s3 - already exists or testing"
+      else
+        aws s3 cp s3://vitessce-data/source-data/vanderbilt_MxIF/vanderbilt.ome.tif "$INPUT/vanderbilt.ome.tif"
+      fi
+      SERVER_URL="https://vitessce-vanderbilt-data.storage.googleapis.com/"
```
Comment (mccalluc) [inline scripts/process_vanderbilt_MxIF.sh]: Would plain`vitessce-data` be better? ... I don't think we should but any effort into a mass migration right now, but I'm not sure having the name this specific helps us?

**Review by mccalluc — COMMENTED** [review]
- Add a note on the Readme about this new data source.
- Add a line in the Readme about installing Google CLI tools... and why we need it.
- Is something like s3_target.txt needed for Google storage?

Ping me for a re-review and I'll read the directions, check it out, and try to run it from scratch. Thanks!

File: scripts/process_vanderbilt_MxIF.sh (hunk 6 above)
Comment (ilan-gold) [reply scripts/process_vanderbilt_MxIF.sh]: Sure, I'll make a new bucket (this was my personal bucket - I forgot to push a branch before I left that has the one Nils made, `vitessce-demo-data`, but I'll change it).

File: fake-files/output-expected/vanderbilt_MxIF/vanderbilt_MxIF.images/vanderbilt_MxIF.tiff.json (hunk 1 above)
Comment (ilan-gold) [reply fake-files/output-expected/vanderbilt_MxIF/vanderbilt_MxIF.images/vanderbilt_MxIF.tiff.json]: I think the two are interchangeable so presumably libvips has some sort of preference that is being revealed here.

File: process.sh (hunk 4 above)
Comment (ilan-gold) [reply process.sh]: Sure, I can change this.

File: process.sh (hunk 2 above)
Comment (ilan-gold) [reply process.sh:42]: In general we probably can't expect this to be universally consistent. They are interchangable.  If you want, I can change the current inputs but we probably want to leave the wildcards.

File: fake-files/output-expected/vanderbilt/vanderbilt.images/vanderbilt.raster.json [hunk 7]
```diff
@@ -0,0 +1,9 @@
+{
+  "width": 50,
+  "height": 50,
+  "tile_size": 512,
+  "channels": {
+    "polyT": "https:/vitessce-data.storage.googleapis.com/vanderbilt.images/polyT.ome.tiff",
+    "nuclei": "https:/vitessce-data.storage.googleapis.com/vanderbilt.images/nuclei.ome.tiff"
```
Comment (mccalluc) [inline fake-files/output-expected/vanderbilt/vanderbilt.images/vanderbilt.raster.json:7]: Do you want to include some kind of version number as part of these URLs?

**Review by mccalluc — APPROVED** [review]
Looks good. I would like to re-run vitessce-data some time... Let me know when a good time / what a good branch is.

**Pull request merged**